File: app/src/main/java/com/iita/akilimo/models/FertilizerPrices.java
package com.iita.akilimo.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.orm.SugarRecord;


@JsonIgnoreProperties(ignoreUnknown = true)
public class FertilizerPrices extends SugarRecord<FertilizerPrices> {

    @JsonProperty("id")
    long id;

    @JsonProperty("recordId")
    int recordId;
    @JsonProperty("priceId")
    int priceId;

    @JsonProperty("minUsd")
    private double minUsd;

    @JsonProperty("maxUsd")
    private double maxUsd;

    @JsonProperty("pricePerBag")
    private double pricePerBag;

    @JsonProperty("active")
    private boolean active;

    @JsonProperty("priceRange")
    private String priceRange;

    @JsonProperty("country")
    private String country;

    @JsonProperty("fertilizerCountry")
    private String fertilizerCountry;

    @JsonProperty("description")
    private String description;

}

File: app/src/main/java/com/iita/akilimo/models/FertilizerPrices.kt
package com.iita.akilimo.models

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import com.orm.SugarRecord
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey

@JsonIgnoreProperties(ignoreUnknown = true)
open class FertilizerPrices : RealmObject() {
    @JsonProperty("id")
    var id: Long = 0

    @JsonProperty("recordId")
    var recordId = 0

    @JsonProperty("priceId")
    var priceId = 0

    @JsonProperty("minUsd")
    var minUsd = 0.0

    @JsonProperty("maxUsd")
    var maxUsd = 0.0

    @JsonProperty("pricePerBag")
    var pricePerBag = 0.0

    @JsonProperty("active")
    var active = false

    @JsonProperty("priceRange")
    var priceRange: String? = null

    @JsonProperty("country")
    var country: String? = null

    @PrimaryKey
    @JsonProperty("fertilizerCountry")
    var fertilizerCountry: String? = null

    @JsonProperty("description")
    var description: String? = null
}